packages/datadog-plugin-aws-sdk/test/dynamodb.spec.js

```diff
@@ -287,247 +287,291 @@ describe('Plugin', () => {
-                      } else {
-                        expect(links[0].attributes).to.deep.equal({
-                          'ptr.kind': DYNAMODB_PTR_KIND,
-                          'ptr.dir': SPAN_POINTER_DIRECTION.DOWNSTREAM,
-                          'ptr.hash': expectedHashes,
-                          'link.kind': 'span-pointer'
-                        })
-                      }
+                  const span = traces[0][0]
+                  const links = JSON.parse(span.meta?.['_dd.span_links'] || '[]')
+
+                  expect(links).to.have.lengthOf(expectedLength)
+
+                  if (expectedHashes) {
+                    if (Array.isArray(expectedHashes)) {
+                      expectedHashes.forEach((hash, i) => {
+                        expect(links[i].attributes['ptr.hash']).to.equal(hash)
+                      })
+                    } else {
+                      expect(links[0].attributes).to.deep.equal({
+                        'ptr.kind': DYNAMODB_PTR_KIND,
+                        'ptr.dir': SPAN_POINTER_DIRECTION.DOWNSTREAM,
+                        'ptr.hash': expectedHashes,
+                        'link.kind': 'span-pointer'
+                      })
                     }
-                    return done()
-                  } catch (error) {
-                    return done(error)
                   }
-                }).catch(error => {
-                  done(error)
-                })
+                  done()
+                }).catch(done)
               })
             }
           }
 
+          // Sets up a test with specific DynamoDB primary keys
+          async function setupTest (primaryKeysJSON) {
+            await agent.close({ ritmReset: false, wipe: true })
+
+            if (primaryKeysJSON) {
+              process.env.DD_TRACE_DYNAMODB_TABLE_PRIMARY_KEYS = primaryKeysJSON
+            } else {
+              delete process.env.DD_TRACE_DYNAMODB_TABLE_PRIMARY_KEYS
+            }
+
+            await agent.load('aws-sdk')
+            AWS = require(`../../../versions/${dynamoClientName}@${version}`).get()
+            dynamo = new AWS.DynamoDB({ endpoint: 'http://127.0.0.1:4566', region: 'us-east-1' })
+          }
+
           describe('1-key table', () => {
-            it('should add span pointer for putItem when config is valid', () => {
-              testSpanPointers({
-                expectedHashes: '27f424c8202ab35efbf8b0b444b1928f',
-                operation: (callback) => {
-                  process.env.DD_TRACE_DYNAMODB_TABLE_PRIMARY_KEYS =
-                    '{"OneKeyTable": ["name"]}'
-                  dynamo.putItem({
-                    TableName: oneKeyTableName,
-                    Item: {
-                      name: { S: 'test1' },
-                      foo: { S: 'bar1' }
+            it('should add span pointer for putItem when config is valid', function (done) {
+              setupTest('{"OneKeyTable": ["name"]}')
+                .then(() => {
```

Comment:
Using async await would probably make these tests more readable.

If you like, we could also briefly pair on this.